jujugui/static/gui/src/app/app.js

```diff
@@ -908,17 +908,16 @@ YUI.add('juju-gui', function(Y) {
           controllerAPI, bakery)}
       />);
 
+      const doCharmstoreLogout = () => {
+        return this.getUser('charmstore') && !this.get('gisf');
+      };
       const LogoutLink = (<window.juju.components.Logout
-        logout={this.logout.bind(this)}
-        clearCookie={bakery.storage.clear.bind(bakery.storage)}
-        gisfLogout={window.juju_config.gisfLogout || ''}
-        gisf={window.juju_config.gisf || false}
         charmstoreLogoutUrl={charmstore.getLogoutUrl()}
-        getUser={this.getUser.bind(this, 'charmstore')}
-        clearUser={this.clearUser.bind(this, 'charmstore')}
+        doCharmstoreLogout={doCharmstoreLogout}
+        locationAssign={window.location.assign}
+        logoutUrl={'/logout'}
```

Comment:
If we're hardcoding it here, why don't we hard code it in the component and save the prop?